application/src/main/java/org/togetherjava/tjbot/commands/free/FreeCommand.java

```diff
@@ -0,0 +1,351 @@
+                logger.debug(
+                        "Channel is not being monitored, ignoring message received in {} from {}",
+                        guildEvent.getChannel().getName(), guildEvent.getAuthor());
+                return;
+            }
+            if (channelMonitor.isChannelBusy(guildEvent.getChannel().getIdLong())) {
+                logger.debug(
+                        "Channel status is currently busy, ignoring message received in {} from {}",
+                        guildEvent.getChannel().getName(), guildEvent.getAuthor());
+                return;
+            }
+            channelMonitor.setChannelBusy(guildEvent.getChannel().getIdLong(),
+                    guildEvent.getAuthor().getIdLong());
+            displayStatus(channelMonitor.getStatusChannelFor(guildEvent.getGuild()));
+            guildEvent.getMessage().reply(UserStrings.NEW_QUESTION.message()).queue();
+        }
+    }
+
+    private @NotNull Optional<Message> getStatusMessageIn(@NotNull TextChannel channel) {
+        if (!channelIdToMessageIdForStatus.containsKey(channel.getIdLong())) {
+            return findExistingStatusMessage(channel);
+        }
+        return Optional.ofNullable(channelIdToMessageIdForStatus.get(channel.getIdLong()))
+            .map(channel::retrieveMessageById)
+            .map(RestAction::complete);
+    }
+
+    private @NotNull Optional<Message> findExistingStatusMessage(@NotNull TextChannel channel) {
+        // will only run when bot starts, afterwards its stored in a map
+
+        Optional<Message> statusMessage = FreeUtil
+            .getChannelHistory(channel, FreeCommandConfig.MESSAGE_RETRIEVE_LIMIT)
+            .flatMap(history -> history.stream()
+                .filter(message -> !message.getEmbeds().isEmpty())
+                .filter(message -> message.getAuthor().equals(channel.getJDA().getSelfUser()))
+                // FIXME the equals is not working, i believe its because there is no getTitleRaw()
+                // .filter(message -> STATUS_TITLE.equals(message.getEmbeds().get(0).getTitle()))
+                .findFirst());
+
+        channelIdToMessageIdForStatus.put(channel.getIdLong(),
+                statusMessage.map(Message::getIdLong).orElse(null));
+        return statusMessage;
+    }
+
+    private void initChannelsToMonitor() {
+        Config.getInstance()
+            .getFreeCommandConfig()
+            .stream()
+            .map(FreeCommandConfig::getMonitoredChannels)
+            .flatMap(Collection::stream)
+            .forEach(channelMonitor::addChannelToMonitor);
+    }
+
+    private void initStatusMessageChannels(@NotNull final JDA jda) {
+        Config.getInstance()
+            .getFreeCommandConfig()
+            .stream()
+            .map(FreeCommandConfig::getStatusChannel)
+            .map(jda::getTextChannelById)
+            // throws NPE if id from config does not match a text channel
```

Comment:
does it still throw? or does the `Objects::requireNonNull` take care of that? does the comment even belong to that line or something else? this kinda confuses me.

Reply:
requiresNonNull throws an NPE?

when we discussed it u said configs are currently designed to fail fast if the config is invalid
so if a config includes an invalid ID for a guild it will not die there
it was previously just being filtered and ignored.

hence the added comment

Reply:
Okay, maybe change the comment to `Config ID must match a text channel` or remove it entirely if you think it doesnt add value.

Reply:
fixed